Source/DafnyLanguageServer.Test/Diagnostics/ListeningTextDocumentLoader.cs

```diff
@@ -0,0 +1,63 @@
+﻿using System.Collections.Generic;
+using System.Linq;
+using JetBrains.Annotations;
+using Microsoft.Boogie;
+using Microsoft.Dafny.LanguageServer.Language;
+using Microsoft.Dafny.LanguageServer.Language.Symbols;
+using Microsoft.Dafny.LanguageServer.Workspace;
+using Microsoft.Extensions.Logging;
+
+namespace Microsoft.Dafny.LanguageServer.IntegrationTest.Diagnostics;
+
+/// <summary>
+/// This class extends the TextDocumentLoader but can record internals for testing purposes:
+/// It currently records the priorities set before every verification round.
+/// </summary>
+public class ListeningTextDocumentLoader : TextDocumentLoader {
+  public List<List<int>> LinearPriorities = new List<List<int>>();
+
+  public ListeningTextDocumentLoader(
+    [NotNull] ILoggerFactory loggerFactory, [NotNull] IDafnyParser parser,
```

Comment:
`#nullable enable` at the top of the file and then you can use `?` instead of `NotNull` annotations?